mlflow/types/schema.py

```diff
@@ -406,3 +425,307 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.inputs)
+
+
+@experimental
+class ParamSpec:
+    """
+    Specification used to represent parameters for the model.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        type: Union[DataType, str],  # pylint: disable=redefined-builtin
+        default: Union[DataType, List[DataType], None],
+        shape: Optional[tuple] = None,
```

Comment:
Reposting https://github.com/mlflow/mlflow/pull/8563/files#r1244551235

Reply:
Oh.. I must reverted this change with some other change